### ISSUE TITLE
Pebble weather for watchface

### ISSUE THREAD

**Issue by danielegobbetti** [Freeyourgadget/Gadgetbridge#482]
The weather data is fetched by the means of [this app](https://f-droid.org/repository/browse/?fdfilter=gelin&fdid=ru.gelin.android.weather.notification), hence the Internet permission is **not** required and is not being added for this functionality to work.

The feature has been released in version 0.17 See the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki/Pebble-System-Apps#weather) to get instructions on how to enable this functionality. The supported watchfaces work on every pebble (including classic), the weather system app is only available on pebbles with firmware 4.

Currently requested watchfaces/apps and statuses:
- [x] System weather app
- [x] [timestyle pebble](https://apps.getpebble.com/en_US/application/55a5c024f4510f794c000071)
- [x] [Square](/turnervink/square)
- [x] [Healthyfy](/prakashait09/Healthify)
- [x] [Fancy](https://github.com/zalewszczak/pebble/tree/master/fancy)
- [x] Tally by [this user](https://github.com/zalewszczak/pebble) but sources not available?
- [x] Crowex by [this user](https://github.com/zalewszczak/pebble) but sources not available?
- [x] [Obsidian](https://github.com/stefanheule/obsidian)
- [ ] [TrekVolle](/aHcVolle/TrekVolle) partial support
- [ ] ~~[Timeboxed](/hussin/timeboxed-watchface)~~ crashes on pebble time
- [ ] [Simply Light](/bhdouglass/simply-light)
- [ ] [DigiMe](/workinghard/DigiMe)
- [ ] [forecal](/clreinki/forecal)
- [ ] [Eight Frame](/aemacleod/eight-frame)
- [ ] [AllThatInfo](/elieserleao/allthatinfo)

**Comment by IzzySoft** [comment]
@danielegobbetti you could have copied our Conversation from yesterday for a starter :) Had I known you wanted it here, I had saved the links on my computer... So please forgive me when I ommit the links to the Pebble Store, and rather focus on links to the resp. Github projects:

* [Timeboxed](/hussin/timeboxed-watchface)
* [Simply Light](/bhdouglass/simply-light)
* [Square](/turnervink/square)
* **[TrekVolle](/aHcVolle/TrekVolle)**
* [DigiMe](/workinghard/DigiMe)
* [forecal](/clreinki/forecal)
* [Eight Frame](/aemacleod/eight-frame)
* **[Healthyfy](/prakashait09/Healthify)**
* [AllThatInfo](/elieserleao/allthatinfo)

(for the books: while in a chat with Daniele on the weather feature, I searched the official Pebble store for FOSS watch-faces with weather support. I might have missed a bunch, but the above results were what looked most promising. The **bold ones** are the ones I'd like to see supported first, bottom-up that is :)

**Comment by ashimokawa** [comment]
@IzzySoft 
Healtify is supported now

**Comment by IzzySoft** [comment]
@ashimokawa Cool! That will be the first thing I try as soon as GB gets the next release :+1: :tada:

**Comment by IzzySoft** [comment]
> The weather data are fetched by the means of [this app](https://f-droid.org/repository/browse/?fdfilter=gelin&fdid=ru.gelin.android.weather.notification)

I've just checked that link. Does that mean we'll have to have those details in the notification area if we want weather on the watch? It's the logical conclusion (and gives me an idea on how you make use of it), but just for clarification it would be nice to say so explicitly :)

**Comment by ashimokawa** [comment]
@IzzySoft
No, you don't have to have the weather in the status bar, just enable Gadgetbridge in the settings as a "skin"

**Comment by danielegobbetti** [comment]
Details will follow with release :-) but the short answer is no, you don't have to have the weather shown in the notification area.
That notification is provided by the weather app as a "skin" that can be disabled.

**Comment by tecufanujacu** [comment]
As first thing happy new year to everyone here.
As second thing can I suggest to also support the weather provider available for cyanogenmod and its derivates? At the moment I'm using the [Yahoo CM Weather Provider](http://www.apkmirror.com/apk/cyanogenmod/yahoo-cm-weather-provider/yahoo-cm-weather-provider-1-0-release/yahoo-cm-weather-provider-1-0-android-apk-download/) but there is also [WU CM Weather Provider](http://www.apkmirror.com/apk/cyanogenmod/wu-cm-weather-provider/wu-cm-weather-provider-1-1-release/wu-cm-weather-provider-1-1-android-apk-download/).

For now i tried to use Weather notification with Gadgedbridge without luck, from what Gadgetbridge version we can use this functionality?

**Comment by ashimokawa** [comment]
This feature is NOT released yet. It is in master but not in any release yet.

**Comment by danielegobbetti** [comment]
I tried timeboxed but it keeps crashing on my watch (without even configuring weather for it).

**Comment by IzzySoft** [comment]
@danielegobbetti maybe skip it for now then. My list was just "things I found", not "things I request" (maybe except for the two bold ones ;)

**Comment by girlwithnoname** [comment]
Would it be possible to support any of Lukasz Zalewski's watchfaces? Most of them seem to be open source (https://github.com/zalewszczak/pebble?files=1). The "fancy" watchface would be cool for example. And the "tally" watchface, too, but I couldn't find the source.

**Comment by danielegobbetti** [comment]
@girlwithnoname will look into it

**Comment by girlwithnoname** [comment]
@danielegobbetti cool, thanks

**Comment by IzzySoft** [comment]
@girlwithnoname That looks like a cool list. Do you know whether there are screenshots and more details on the projects listed? And a place where to obtain the related `.pbw` files?

**Comment by girlwithnoname** [comment]
@IzzySoft Most of them if not all can all be found in the Pebble appstore. That's where I know them from. Here's the link: https://apps.getpebble.com/en_US/developer/5299dfb9129af7a225000093/1?dev_settings=true

**Comment by IzzySoft** [comment]
@girlwithnoname Thanks! Now I know why I missed them on my hunt: didn't notice the (small) weather :)

@danielegobbetti Good addition to my results. I mainly had the digital clocks covered, and these are all analogue ones :)

**Comment by danielegobbetti** [comment]
They seem to share the same weather message fortunately. I added support for the two requested by @girlwithnoname and crowex because I like it :-)

**Comment by girlwithnoname** [comment]
Thanks @danielegobbetti and @IzzySoft :-)

**Comment by tecufanujacu** [comment]
@ashimokawa and @danielegobbetti 
just to know, what do you think about the idea of using the weather provider integrated in the cyanogenmod? ([#482,269887021](https://github.com/Freeyourgadget/Gadgetbridge/issues/482#issuecomment-269887021))

**Comment by danielegobbetti** [comment]
@tecufanujacu I can't do much about it as I am using  plain AOSP on my android device.

**Comment by tecufanujacu** [comment]
Understood, anyway if it can help the code of the two provider is available on the github:
- [here the code for Yahoo CM Weather Provider](https://github.com/CyanogenMod/android_packages_apps_YahooWeatherProvider)
- [here the code for WU CM Weather Provider](https://github.com/CyanogenMod/android_packages_apps_WundergroundWeatherProvider)

I leave the link here just for reference.

**Comment by ajiekc905** [comment]
WeatherGraph by Tomas Kafka ?

**Comment by IzzySoft** [comment]
@ajiekc905 link to app and source?

**Comment by danielegobbetti** [comment]
FYI, we tagged yesterday and the new version will be available on fdroid soon (hopefully).

**Comment by IzzySoft** [comment]
Congrats! So we still keep this open for requests on other watch faces – or do you prefer a separate issue for each?

**Comment by tecufanujacu** [comment]
I don't have "weather app (system)", I think because I have never used the official app, is there something that I can do?

**Comment by jgkamat** [comment]
@tecufanujacu I'm seeing the same problem (and weather isn't working for me either). I'm using the pebble   v1_5 with firmware 3.12.3 if that makes a difference. I tried flashing with the official pebble app and even with that no weather app showed up.

**Comment by IzzySoft** [comment]
@tecufanujacu enable it from within GB? Worked great for me: tap your device (to enter the app manager), long-tap the weather system app, tap "enable".

As for weather support: when long-tapping that weather app without having the Android weather notification app installed, it will prompt you to do so. Then, in that one, make sure to have the "Gadgetbridge" skin enabled. After that, weather should work in supported watch faces.

Will update the wiki ([Pebble System Apps](https://github.com/Freeyourgadget/Gadgetbridge/wiki/Pebble-System-Apps)) accordingly as soon as time permits.

**Comment by danielegobbetti** [comment]
@tecufanujacu @jgkamat you are both right, there is no weather system app on fiirmware <4. However, that app is not needed, it's just the place where we put the shortcut to enable the weather function.

So on firmware <4 you have to manually ensure the weather notification app is installed ( https://f-droid.org/repository/browse/?fdid=ru.gelin.android.weather.notification ) and then launch it and ensure that the gadgetbridge "skin" is enabled.

**Comment by IzzySoft** [comment]
Thanks for the correction, @danielegobbetti – I've just updated the wiki for [weather](https://github.com/Freeyourgadget/Gadgetbridge/wiki/Pebble-System-Apps#weather) :innocent:

**Comment by danielegobbetti** [comment]
Thanks a lot for the contribution to the wiki!

**Comment by IzzySoft** [comment]
Anytime, @danielegobbetti – I somehow feel responsible for the Pebble section :innocent:

**Comment by tecufanujacu** [comment]
With the watchface TimeStyle the unit goes from from celsius to fahrenheit when the pebble reconnect after disconnect, I reported the problem in the issue [#105](https://github.com/freakified/TimeStylePebble/issues/105) of TimeStyle or this could depend from GadgetBridge?

**Comment by danielegobbetti** [comment]
Mhh that is strange because timestyle is my daily driver watchface and I don't have this problem. Can you please share the version of the watchface you are using?

**Comment by sunng87** [comment]
Timeboxed now works for me except weather doesnt update. Possible to recheck the status? Thank you!

**Comment by krayon** [comment]
This should maybe be in a separate Issue but I'm wondering is there a standard/recommended way to add weather to a watch face to work with GadgetBridge without requiring GadgeBridge modification?

I have only coded stand alone faces until now so the question may not make sense - I haven't looked into how the watch to phone comms work. I would however like to develop a new watch face and have weather data. I don't care about Pebble App compatibility really, just Gadget Bridge.

**Comment by danielegobbetti** [comment]
Yes we have to open a separate issue for this, but we have some plans. I'll share more details in a new issue soon. Using openweathermap as data model is a safe bet anyway :-)

**Comment by danielegobbetti** [comment]
@krayon I created the new issue, let's move there for the discussion. I have also cleaned up a bit the thread on github.

**Comment by danielegobbetti** [comment]
@ozdreamern obsidian support is now in master and will be in the next release.

**Comment by pablob127** [comment]
Another nice open source watchface to support could be [Minimalin](https://apps.getpebble.com/en_US/application/56f93a5361a01637e5000036), source at [GitHub](https://github.com/groyoh/minimalin). 

Although it seems to me that open source faces would be best supported by deciding on an API so that the watchface developers can add the support (non-open source watchfaces could be supported with special handlers). Too bad I know very little about Android or Pebble develoment to help on that...

**Comment by eevleevs** [comment]
Can you support AWW1?

**Comment by Denster1964** [comment]
I enjoy the obsidian watchface, if I can get the temperature to read in Fahrenheitwould be great. I have it set to Fahrenheit in weather notification and the  Timeline reads correctly but watch face and system weather reads Celsius.

**Comment by Denster1964** [comment]
Looks like the CONFIG_WEATHER_UNIT_LOCAL value isn't being passed, should be 1 for Celcius or 2 for Fahrenheit.  Like the app, pretty awesome project.

**Comment by danielegobbetti** [comment]
@Denster1964 yes celsius is forced ATM. Since the background-js branch uses the original code of each watchface (no special handler needed anymore) your problem will be fixed when we release that.

**Comment by Denster1964** [comment]
Sounds good, doing a great job

**Comment by chaseadam** [comment]
Any advice for troubleshooting no weather on Obsidian watchface?
* Older watch with 3.12.3 FW (latest for the original pebble?), so no access to "weather" app in gadget bridge
* Weather Notification installed with gadget bridge skin enabled
* Running latest from FDroid (3.17.5)

**Comment by danielegobbetti** [comment]
Perhaps we could add an indicator of the weather data status (availability, last update time, etc). If you launch the weather app on your android you can force the update by pushing the "reload" button, as it triggers a sync with all configured skins as well.

**Comment by Denster1964** [comment]
sounds good

**Comment by ianmcxa** [comment]
I would like to create a watchface compatible with Gadgetbridge's weather. I'm seeing that for every supported watchface, custom code is added to GadgetBridge. I could just do that for my watch face, but it would be nice to have an API for weather. Is there still interest in making one?

**Comment by danielegobbetti** [comment]
@ianmcxa this branch https://github.com/Freeyourgadget/Gadgetbridge/tree/background-javascript exposes the raw responses from openweathermap to existing watchfaces in a completely transparent manner (read: there is no need to write custom code for the watchfaces, gadgetbridge reconstruct the server response and provides it to the watchface JS). If you can contribute to it I'd appreciate it very much.

A different / non provider-dependent API could be discussed in issue #528, if you think it's useful.

**Comment by bigrjsuto** [comment]
Would love if MultiFit could be supported!

**Comment by danielegobbetti** [comment]
Multifit works already in the background js branch with  (it's my daily driver lately)

**Comment by bigrjsuto** [comment]
Then maybe it's something I'm doing wrong? I can only seem to get the Square and Healthify watchfaces to display weather information.

**Comment by danielegobbetti** [comment]
*If you are using the background js branch* (this means you are not running the fdroid distributed application, but you built yourself the app from sources using the aforementioned branch[0]), multifit should be working and if it doesn't please open a new issue and attach logs.

[0] sorry if I sound patronizing but I have no idea of your level of expertise and must make sure you understand what I mean when I say "branch" etc.

**Comment by bigrjsuto** [comment]
I'm not using the js branch. I do not know how build f-droid myself. Is there any kind of tutorial you could link me?

Thank you very much, btw.

**Comment by yaomtc** [comment]
I've tried Healthify, Square, and TimeStyle, but none of them show the weather. I've installed Weather Notification, enabled Gadgetbridge in its settings, activated the Weather system app, and now the weather is shown inside the Pebble weather app. However none of the three watchfaces I tested show it. Any ideas?

**Comment by danielegobbetti** [comment]
Did you configure the watchfaces to show the weather? At least timestyle doesn't show weather unless you enable it.

**Comment by IzzySoft** [comment]
Healthify shows weather as well, I've tried that one here right when it was implemented. Healthify was among the first ones to get weather support from GB, together with TrekVolle.

**Comment by bbenne10** [comment]
@bigrjsuto: You can find documentation on how to build the application from source [here](https://github.com/Freeyourgadget/Gadgetbridge/wiki/Developer-Documentation). This doc describes setting up everything (with the possible exception of the Android SDK; I'm not sure how to install that on Debian and I think this might not cover it...) 

(Sidenote since we're discussing the branch here: The `background-javascript` branch looks good to me for the most part, but there seem some latent bugs (I get `{AppKeyWeatherFailed: 0}` as my configuration for Minimalin every time I configure it - it seems to "flash" the old config, but switch over to this message before I can hit send. I've looked at logs, but haven't been able to find anything relevant. I'll open a new ticket when I have more information.)

**Comment by yaomtc** [comment]
That issue went away on its own somehow. But now I have another issue: it won't stay on Fahrenheit. Weather Notification shows its Fahrenheit setting correctly, which I thought Gadgetbridge pulls data from. However the Pebble's system weather app shows Celsius, and when I configure a skin, choose Fahrenheit, send the data over, it's Fahrenheit temporarily, but soon switches back to Celsius. Should I create another issue for this?

**Comment by danielegobbetti** [comment]
> Should I create another issue for this?
@yaomtc no, see above in this very thread.

**Comment by witheld9** [comment]
This doesn't seem to be working in any watchfaces on my Pebble Time

**Comment by grimsrud** [comment]
Would it be possible to support weather-my-way of Jared Biehler (https://github.com/jaredbiehler/weather-my-way)?

**Comment by pebblebro** [comment]
Hi everyone I'm a couldpebble dev I have good news for watchfaces you can get weather data by offloading to a another server by using this command in cloud pebble when your making the watch face pebble://custom-boot-config-url/CUSTOM_URL

**Comment by foyaxe** [comment]
I would be glad to see support for my favourite watchface:
Multifunctional 7 Segment
https://github.com/cfg1/pebble-mss

**Comment by djibux** [comment]
Hello, I use TimeStyle and I have the feeling the forecast complication displays the min-max temperature for the day, but the icon represent tomorrow's weather.  This is somehow confusing.  Could the forecast icon be set to today's weather?

**Comment by icovelev** [comment]
This watchface is nice. I would love to see a support for it. (Aspire by Elijah Kang)
https://apps.getpebble.com/en_US/application/5798b1b81482cd673b0000a2?dev_settings=true&native=false&query=Aspire&section=watchfaces

**Comment by IzzySoft** [comment]
@icovelev afraid it's not open source, so chances are rather low IMHO.

**Comment by sethron** [comment]
I did get weather to work with Pebble Time. I had to install version 0.38 then upgrade to 0.39 of Weather notification from f-droid to get it to play nice. Perhaps we can add it to the wiki?

**Comment by VmKid** [comment]
I'd like to see support for Pebble Mario:

https://github.com/ClusterM/pebble-mario

**Comment by ZeroBit** [comment]
Is there any chance to have weather support for openweathermap.org for very popular watchface timely https://github.com/cynorg/PebbleTimely
Here is the fragment from PebbleTimely/src/js/pebble-js-app.js from lines 414, 415
`req.open('GET', "http://api.openweathermap.org/data/2.5/weather?" +
    "lat=" + latitude + "&lon=" + longitude + "&appid=fdc43ca42ea6a45d9c73a810f840aa55" + "&cnt=1" + "&units=" + units, true);`

Usually I put my own openweathermap ID here and it works great

On the basis of Timely this "Time and Calendar" watchface is made. It is here. 
https://github.com/UnnamedHero/pebble-watchface-time-and-calendar
It allows in settings to use own openweathermap ID. 

But unfortunately none of these outstanding watchfaces show weather in gadgetbridge. Please, fix it.

**Comment by ajiekc905** [comment]
Tried to install all available on f-droid Weather notification apps and all of them telling: 
Weather update failed: For input string "-33.XXXX, 151.XXXX"

**Comment by bimbar** [comment]
Timeboxed works, but only in Fahrenheit. Celsius still displays the Fahrenheit numbers.

**Comment by enFactorial** [comment]
I would like to add my voice for weather support for Multifunctional 7 Segment watchface. This is an amazingly informative and easily readable watchface that recently got a very useful update.

https://github.com/MarSoft/pebble-mss/releases/tag/v15.7.0-mod2-beta.1

**Comment by Gabicu** [comment]
I think taht Multifunctional 7-Segment is a complet and good weatherface to pebble. If you get it to work with GB, great. 😉
I usse classic pebble.

**Comment by witheld9** [comment]
@danielegobbetti why was this closed?

**Comment by ashimokawa** [comment]
This was closed because in theory weather should works without special support when background JavaScript is used

**Comment by IzzySoft** [comment]
… and not crashing right ahead, @ashimokawa :wink: I cannot use that with *any* watch face, remember? Issue is still open (AFAIR Daniele works on it, but I haven't heard back for a while).

**Comment by ashimokawa** [comment]
Webview crashing is only known on vintage marshmallow devices. Its not that it crashes for everyone 😇

**Comment by pasbec** [comment]
For me, backgroud JS does not seem to work with any Watchfaces. After enabling it an my Galaxy Note 3 with the latest LOS 14.1, as soon as I try to reconnect to the Pebble 2, Gadgetbridge just goes black and unresponsive. After forcibly killing Gadgetbridge, connection to the Pebble 2 suddenly works, however e.g. TrekVolle does not update the weather. The phone battery state is shown correctly though and the Pebble Weather App correctly shows the weather with the Gadgetbridge skin in weather notification.

**Comment by IzzySoft** [comment]
@ashimokawa I didn't say that. But neither I'm the only one. Plus, as you can see by the previous post, trouble isn't limited to "vintage marshmallow devices" (LOS 14 is Nougat).

@pasbec Yepp, that issue with TrekVolle is known. Wasn't it @ashimokawa who had it as well? Or was that Carsten? So at least one of the devs can reproduce that. And yes, battery is what I'm missing as well in TrekVolle.

**Comment by ashimokawa** [comment]
Since JavaScript seems to be unusable for many for unknown reasons, I reopened this

**Comment by tecufanujacu** [comment]
@ashimokawa and @danielegobbetti 

A question, is it now possible to update the weather with the Cyanogenmod Weather providers as I proposed time ago (#[482,269887021](https://github.com/Freeyourgadget/Gadgetbridge/issues/482#issuecomment-269887021) / #[482,271617008](https://github.com/Freeyourgadget/Gadgetbridge/issues/482#issuecomment-271617008)) instead of using the app Weather notification?

I have just saw [CMWeatherReceiver.java](https://github.com/Freeyourgadget/Gadgetbridge/blob/master/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/CMWeatherReceiver.java) and the new [support for Omnirom Omnijaws weather service](https://github.com/Freeyourgadget/Gadgetbridge/commit/888e7fa2292ef37092172685e5f5926ef9d6fc42#diff-793b216d5a942ac8972907fc2f96c003). 

I tried uninstalling the app Weather notification but the watchfaces aren't able to update the weather.

**Comment by Gabicu** [comment]
I think taht Multifunctional 7-Segment and NineTime are complet and good weatherfaces to pebble. If you get it to work with GB, great. 😉
I usse classic pebble. And 2018 is coming and the pebble's sevidores stop, and we don't know what will happen.

**Comment by tedrock** [comment]
[Slides of Time](https://apps.getpebble.com/en_US/application/555283c7efff0cc8e50000b0) please

**Comment by Gabicu** [comment]
In Trekvolle, Timestyle y AllThatInfo don't works the weather to me. Canvas don't work the weather, only if I open Canvas and touch the face, the weather change in pebble

**Comment by Gabicu** [comment]
Well, Trekvolle wok. 
I think that TimeStyle work.
AllThatInfo don't work

**Comment by asardaes** [comment]
I have a couple of watchfaces that use OWM (as shown in the pebble tutorials), and I'd like to make them work with gadgetbridge. If the background javascript stuff works, should they work automatically? Or is there still something I'd need to modify?

**Comment by ashimokawa** [comment]
They should just work with background JavaScript.


Given that you either have weathernotification installed or lineage OS with a weather provider of your choice (I tested the yahoo provider)

This is where we get weather data from

**Comment by bockchristian** [comment]
I would love support for http://bestwatchapps.net/app/pebble/make-me-smile/

Thanks for your great work!

**Comment by ashimokawa** [comment]
@bockchristian 
Have you tried background javascript?

**Comment by Mathes75** [comment]
I would love to use weather in https://github.com/PanicMan/Pebble-Casio-WV58DE

Great work in general!

**Comment by Gabicu** [comment]
Multifunctional 7-Segment and NineTime, please. Thank's

**Comment by danielegobbetti** [comment]
@Mathes75 this looks like an error on our side, I will look into it!

**Comment by Mathes75** [comment]
Cool! Many thanks!

**Comment by danielegobbetti** [comment]
@Mathes75 the issue is fixed now in master, thanks for the precise bug report! If you are using a self-compiled version, you can update and build master, otherwise you have to wait for the next release.
However the weather was working for me with this watchface also before the fix, so if you don't see it the issue might reside somewhere else. I suggest you open a new issue detailing the device and android version you are using (e.g. LineageOS or Omnirom based).

@Gabicu this is not a support forum. You can either try adding support for your desired watchfaces by yourself (and ask for help if something is not clear), or at the very minimum provide the basic details of your current setup.

**Comment by Mathes75** [comment]
Thanks @danielegobbetti!  Will give it a try with the next release. I'm quite new to Pebble and Gadgetbridge and don't know how to compile (yet).

**Comment by czandor** [comment]
Hello,

I am a developer of NineTime watchface.
I started to make compatible my watchface with gadgetbridge.
Now, almost all the function is working (in not pubilc version), but the current weather wind (speed), and the forecast weather is missing from the openweathermap gatweay.
And it would be good if we could find something for my info script.
Here is a help for my info script soltuion: http://czandor.hu/pebble/info-help.php

**Comment by danielegobbetti** [comment]
@czandor I am now testing your watchapp (nice work btw! I wonder if you would consider open sourcing it, but I digress :-) ).

Regarding the forecast, how many days do you need? We get 5 days from the LineageOS/Omnirom weather providers, but only 3 from the weather notification app (that's why we are not mimicking the forecast request ATM). I will check if wind speed is available and add it to our JSON.

**Comment by czandor** [comment]
@danielegobbetti I'm using the most available days to calculate the expected min and max temperature.
My plan is to make it opensource after I cleaned the code (there is many private apikeys in code), maybe with version 2.00
Thanks for helping.

**Comment by ZehHa** [comment]
As I'm using a Pebble 2 HR, I would like to get weather information on the HR Vitals watchface. Are there any plans to support that particular watchface?

**Comment by foosinn** [comment]
Hey,

anyone using Timestyle and still gets weather updates? Mine is now stuck for quite a while. The weather in the watch gets correctly updated.

**Comment by czandor** [comment]
@danielegobbetti Please can you put the wind data to OWM weather json? And it is expected that weather forecast will be included for at least for 3 days?
I want to update my app with gb compatibility fix, but now I have to disable the wind and forecast now. Anyway, maybe you have to send the gb version number to pebble config site, because I need to know which feature must disable.

**Comment by Gabicu** [comment]
With Timestyle, the same thing has happened to me for a long time, a pity

**Comment by Gabicu** [comment]
Please, a face with  the wind data? Thank's

**Comment by danielegobbetti** [comment]
@czandor next release will contain wind data. :-) Thanks for the heads up! I also fixed a long standing bug thanks to adapting the mimicked jSON.

**Comment by casterland** [comment]
Would be very grateful to see Forcaswatch working.  I believe it uses wunderground API calls

**Comment by teresmajor** [comment]
Hi - I'm also looking for forcaswatch: https://apps.getpebble.com/en_US/application/558d4139df1af0344a00008f?query=forcaswatch&section=watchfaces
Is there any chance of keep it working with gadgetbridge and weather notification?

**Comment by IzzySoft** [comment]
Is the source code available? If not, it looks bad.

**Comment by teresmajor** [comment]
I don't think so - just the .pbw-file on the Pebble Server. And I don't know, who the developer "RCY" is, or how to get in contact with him.

**Comment by Gabicu** [comment]
Some news from TimeStyle

**Comment by jfdhuiz** [comment]
Would love to get this working for Pebble Multifunctional Seven Segment (aka m7s).

https://github.com/cfg1/pebble-mss

**Comment by pbhgl** [comment]
Hopefully this is the right place to ask? As a watchface developer it would be interesting to know

* what weather values are available (weather condition, wind speed / direction, temperature, current weather, daily forecast, hourly forecast etc.)
* what adjustments are necessary for the watchface to display weather information with Gadgetbridge?

Background info: I am currently working on a new watchface as a successor of [AWW1](https://apps.rebble.io/en_US/application/568ec32dfc66e37c3000006c) and would like to make it working for Gadgetbridge users, too.

**Comment by danielegobbetti** [comment]
@pbhgl as far as I can tell the weather info is correctly reconstructed, forecast is not available/reliable at all (hourly nor daily).

You don't need any adjustment in your watchapp/watchface if it supports openweathermap as a source. Weather will be shown only for users who have background js turned on.

**Comment by LoneFenris** [comment]
Not sure where else to put this, but I notice that the Futura watchface ([Rebble store](http://store.rebble.io/app/52d313a912ea3dcf84000024), [source code](https://github.com/Katharine/futura-weather-sdk2.0/tree/pebble-fixes)) is not configurable as the attempt throws an Uncaught TypeError.

**Comment by danielegobbetti** [comment]
@LoneFenris the link to the store returns a 404

**Comment by jcrode** [comment]
> Would love to get this working for Pebble Multifunctional Seven Segment (aka m7s).
> 
> https://github.com/cfg1/pebble-mss

That's pull request  #1371.

**Comment by danielegobbetti** [comment]
@jcrode #1371 has been merged, thanks a lot.

**Comment by jcrode** [comment]
@danielegobbetti I've added support for two more watchfaces with #1383 , thanks!

**Comment by jumper047** [comment]
Hi! Can you add support for this watchapp?
https://github.com/stefanheule/graphite

**Comment by w2helm** [comment]
I'd like to second the request for ForcasWeather.  There is no app like it.  It offers graph of temperature, solar and moisture for the upcoming 24 hours, it works with Weather Underground and has a calendar.  And it just started reporting no data a couple days ago (survived the Rebble switch and everything).  But I have no idea of who the source is or how to contact them.

**Comment by mattrossman** [comment]
@w2helm I've been having the same issue, the Weather Underground API site says they are shutting the free service down. Perhaps that is the issue here (though I thought mine was still working after their announced end of service date)

**Comment by w2helm** [comment]
Thanks for the information.  I have a WU account, but it was free.  Maybe I need to talk to them.

 

Thanks,

Wilton
